bftengine/src/bftengine/KeyManager.h

```diff
@@ -1,47 +1,50 @@
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License"). You may not use this product except in
+// compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright notices and license terms. Your use of
+// these subcomponents is subject to the terms and conditions of the sub-component's license, as noted in the LICENSE
+// file.
+
 #pragma once
 
-#include "Serializable.h"
 #include "InternalBFTClient.h"
+#include "KeyStore.h"
 
 class KeyManager {
  public:
-  struct KeyExchangeMsg : public concord::serialize::SerializableFactory<KeyExchangeMsg> {
-    std::string key;
-    std::string signature;
-    int repID;
-    KeyExchangeMsg(){};
-    KeyExchangeMsg(std::string key, std::string signature, int repID);
-    std::string toString() const;
-    static KeyExchangeMsg deserializeMsg(const char* serStr, const int& size);
-
-   protected:
-    const std::string getVersion() const;
-    void serializeDataMembers(std::ostream& outStream) const;
-    void deserializeDataMembers(std::istream& inStream);
-  };
+  static KeyManager& get(InternalBFTClient* cl = nullptr, const int id = 0, const uint32_t clusterSize = 0) {
```

Comment:
The attribute `const` is redundant for the arguments passed by value.

Reply:
Disagree, const is meant for the reader as for the compiler. it doesn't matter if it has a side effect. it tells the reader that this variable shouldn't change.
Furthermore, inside the function one can change the value by mistake., and initialize a member with it for example.

Reply:
The reader cannot change the passed variable anyways, since it is passed by value.
The fact that the variable is passed by value makes it clear to reader that it cannot be changed(since it is copied).
The copied variable can be changed by mistake:
```
auto id1 = id;
++id;
```
If we take a look at various functions and classes of the standard library, primitive types are always passed w/o const and references:
https://en.cppreference.com/w/cpp/container/vector/resize

Reply:
the reader doesn't know that the function can't change the value inside. which he shouldn't.
const tells him that. as I said const is for the reader to understand the code by looking at the signature.
without the const I lose information, why does it bother you?
and the standard is not a good example, it's not an open source product.

Reply:
Ok man, if C++ standard library is not a good example of C++ code and its code is not an open-source project I am afraid this conversation is meaningless.
P.S. [GCC](https://github.com/gcc-mirror/gcc/tree/a1c9c9ff06ab15e697d5bac6ea6e5da2df840cf5/libstdc%2B%2B-v3), [clang](https://github.com/llvm-mirror/libcxx/tree/78d6a7767ed57b50122a161b91f59f19c9bd0d19).

Reply:
let's open a file in the standard library and see home much code is understandable.
I happened to look at it quite a lot